src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(state, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach(element => {
+        delete state[element];
+      });
+    }
+
+    if (action.type === 'clear') {
+      for (const stat in state) {
```

Comment:
```suggestion
      for (const key in state) {
```
inside `for in` loop you have a  key of each property, so it better name it `key` explictly